lms/static/js/commerce/views/receipt_view.js

```diff
@@ -19,6 +19,11 @@ var edx = edx || {};
             this.useEcommerceApi = this.ecommerceBasketId || this.ecommerceOrderNumber;
             _.bindAll(this, 'renderReceipt', 'renderError', 'getProviderData', 'renderProvider', 'getCourseData');
 
+            /* Mix non-conflicting functions from underscore.string (all but include, contains, and reverse) into
+             * the Underscore namespace.
+             */
+            _.mixin(_s.exports());
```

Comment:
Removing this was the key to the bug that was introduced, I believe. It is a shame that tests didn't catch the problem, so it is hard to know for sure that the revert is the right solution. @mattdrayer can you try this out and see if it solves the particular bug you were seeing?